libs/go/sia/aws/agent/agent.go

```diff
@@ -525,6 +543,7 @@ func RunAgent(siaCmd, siaDir, ztsUrl string, opts *options.Options) {
 						errors <- fmt.Errorf("refresh identity failed: %v\n", err)
 						return
 					}
+					identityRefreshed <- true
```

Comment:
this seems to be in the wrong place and will cause a hang in the agent since we can't keep writing to the channel if no-one is reading it. e.g. if we have no token setup, after the initial setup, during the 2nd this this call will hang since the initial write hasn't been read anyone. 

we should make this after line 512 after we call RegisterInstance. Even then we should only do if we have valid token options configured (similar check like on line 590).